### ISSUE TITLE
Add support to create an EC private key from Curve NID and BIGNUM scalar only

### ISSUE THREAD

**Issue by bgrieder** [sfackler/rust-openssl#2075]
The function `int EC_KEY_set_private_key(EC_KEY *key, const BIGNUM *prv);` is available in the openssl API but is not correctly exposed in rust-openssl.

The available function is 
```Rust
/// Constructs an public/private key pair given a curve, a private key and a public key point.
    #[corresponds(EC_KEY_set_private_key)]
    pub fn from_private_components(
        group: &EcGroupRef,
        private_number: &BigNumRef,
        public_key: &EcPointRef,
    ) -> Result<EcKey<Private>, ErrorStack>;
```
But this one requires a public key, which is not always available and does not match the `EC_KEY_set_private_key` parameters.

**Comment by Skepfyr** [comment]
Yeah this method somewhat weirdly corresponds to `EC_KEY_set_group`, `EC_KEY_set_private_key`, and `EC_KEY_set_public_key`. If you need them we'd happily accept a PR to add a method that invokes just one of those OpenSSL functions.
